python_modules/dagster/dagster/_grpc/impl.py

```diff
@@ -274,42 +275,44 @@ def get_external_schedule_execution(
 ):
     from dagster._core.execution.resources_init import get_transitive_required_resource_keys
 
-    schedule_def = repo_def.get_schedule_def(schedule_name)
-    scheduled_execution_time = (
-        pendulum.from_timestamp(
-            scheduled_execution_timestamp,
-            tz=check.not_none(scheduled_execution_timezone),
+    try:
+        schedule_def = repo_def.get_schedule_def(schedule_name)
+        scheduled_execution_time = (
+            pendulum.from_timestamp(
+                scheduled_execution_timestamp,
+                tz=check.not_none(scheduled_execution_timezone),
+            )
+            if scheduled_execution_timestamp
+            else None
         )
-        if scheduled_execution_timestamp
-        else None
-    )
 
-    required_resource_keys = get_transitive_required_resource_keys(
-        schedule_def.required_resource_keys, repo_def.get_top_level_resources()
-    )
-    resources_to_build = {
-        k: v for k, v in repo_def.get_top_level_resources().items() if k in required_resource_keys
-    }
-
-    with ScheduleEvaluationContext(
-        instance_ref,
-        scheduled_execution_time,
-        repo_def.name,
-        schedule_name,
-        resources=resources_to_build,
-    ) as schedule_context:
-        try:
-            with user_code_error_boundary(
-                ScheduleExecutionError,
-                lambda: "Error occurred during the execution function for schedule {schedule_name}".format(
-                    schedule_name=schedule_def.name
-                ),
-            ):
+        required_resource_keys = get_transitive_required_resource_keys(
+            schedule_def.required_resource_keys, repo_def.get_top_level_resources()
+        )
+        resources_to_build = {
+            k: v
+            for k, v in repo_def.get_top_level_resources().items()
+            if k in required_resource_keys
+        }
+
+        with user_code_error_boundary(
+            ScheduleExecutionError,
+            lambda: "Error occurred during the execution function for schedule {schedule_name}".format(
+                schedule_name=schedule_def.name
+            ),
+        ):
+            with ScheduleEvaluationContext(
```

Comment:
[1] we should leave a note here about why this is in the user code boundary since its not immediately obvious

`# user code to build resources potentially run when creating/entering this context`